text/0000-command-ergonomics.md

```diff
@@ -342,6 +344,13 @@ impl SubprocessError {
     // and a caller can't reimplement it without making assumptons
     // about `SubprocessError`'s contents.
     fn has_problem(&self) -> bool;
+
+    /// Returns `Ok<ExitStatus>` if the only reason for the failure was a nonzero exit status.
+    /// Otherwise returns `self`.
+    ////
+    /// Use this if you want to to tolerate some exit statuses,
+    /// but still fail if there were other problems.
+    pub fn just_status(self) -> Result<ExitStatus, SubprocessError>;
```

Comment:
This feels a little awkward to use: if you call this to filter out some exit statuses, you lose the rest of the error, which means you don't have `stdout_bytes` anymore. What code pattern would you suggest for code that wants to run `diff`, check the exit code for 0/1 to determine if there was a difference, bail on >1, and capture the patch output?

Reply:
I wonder if this (straw proposal) might be easier to use:

```rust
/// If the process is only considered to have errored due
/// to its exit code, calls the provided filter, returning
/// `Ok(Self)` if it returns `true`, or `Err(Self)` if it
/// returns `false`.
pub fn permit_statuses(self, f: impl FnOnce(ExitStatus) -> bool) -> Result<ProcessError, ProcessError>
```

This would make it relatively straightforward to say "allow 0 or 1, reject everything else". (Still mildly annoying due to `code()` returning `Option`, but doable.)

Reply:
I quite like this idea.  I wasn't sure people would like the notion of returning `Ok(ProcessError)`, but if you think that's OK it seems OK to me.

>  mildly annoying due to `code()` returning `Option`

I wondered whether the closure should receive `ExitCode` (so would only be called if the process exited rather than was killed).  But then you can't use this to ignore (say) `SIGPIPE`.

I do kind of hate booleans.  Would `impl FnOnce(ExitStatus) -> Result<(), ()>` be too awful?